bin/ap_proto

```diff
@@ -0,0 +1,337 @@
+
+        return objects
+
+    def _forcedPhotometry(self, objects, latest_objects, dt):
+        """
+        Do forced photometry on latest_objects which are not in objects.
+
+        Extends objects list with new DiaObjects.
+        """
+
+        # Ids of the detected objects
+        ids = set(obj.diaObjectId for obj in objects)
+
+        for obj in latest_objects:
+            # only do it for 30 days after last detection
+            if obj.diaObjectId in ids: continue
+            delta = dt - obj.lastNonForcedSource
+            if delta > timedelta(days=30): continue
+
+            # make new object with the same ID
+            new_obj = l1db.DiaObject(diaObjectId=obj.diaObjectId, validityStart=dt, validityEnd=None,
+                                     lastNonForcedSource=obj.lastNonForcedSource,
+                                     ra=obj.ra, decl=obj.decl, raSigma=0., declSigma=0., ra_decl_Cov=0.,
+                                     muRa=0., muRaSigma=0., muDecl=0., muDecSigma=0., muRa_muDeclCov=0.,
+                                     parallax=0., parallaxSigma=0., muRa_parallax_Cov=0., muDecl_parallax_Cov=0.,
+                                     lnL=0., chi2=0., N=0,
+                                     flags=0, htmId20=obj.htmId20)
+            objects.append(new_obj)
+
+    def _makeDiaSources(self, sources, indices, visit_id):
+        """
+        Generate set of DiaSources to store in a database
+        """
+
+        srcs = []
+        for i in range(len(sources)):
+
+            xyz = sources[i]
+            var_idx = indices[i]
+
+            v3d = Vector3d(xyz[0], xyz[1], xyz[2])
+            ra = LonLat.longitudeOf(v3d).asDegrees()
+            decl = LonLat.latitudeOf(v3d).asDegrees()
+            dir_v = UnitVector3d(v3d)
+            index = htmIndex(dir_v, constants.HTM_LEVEL)
+
+            self.lastSourceId += 1
+            src = l1db.DiaSource(diaSourceId=self.lastSourceId, ccdVisitId=visit_id,
+                                 diaObjectId=var_idx, filterName='x',
+                                 prv_procOrder=0, midPointTai=0.,
+                                 ra=ra, decl=decl, raSigma=0., declSigma=0., ra_decl_Cov=0.,
+                                 x=0., xSigma=0., y=0., ySigma=0., x_y_Cov=0., snr=5.,
+                                 flags=0, htmId20=index)
+            srcs += [src]
+
+        return srcs
+
+
+    def _makeDiaForcedSources(self, objects, visit_id):
+        """
+        Generate set of DiaForcedSources to store in a database
+        """
+
+        fsrcs = []
+        for obj in objects:
+
+            src = l1db.DiaForcedSource(diaObjectId=obj.diaObjectId, ccdVisitId=visit_id,
+                                       psFlux=0., psFlux_Sigma=0.,
+                                       x=0., y=0.,
+                                       flags=0)
+            fsrcs += [src]
+
+        return fsrcs
+
```

Comment:
In first approximation it should not matter, and probably even at second :neutral_face: